Setup.hs

```diff
@@ -29,36 +24,28 @@ import Distribution.Simple.Setup
 import Distribution.Simple.UserHooks
   ( UserHooks (buildHook, confHook, cleanHook)
   )
-import Distribution.Simple.Utils (rawSystemExit)
+import Distribution.Simple.Utils
+  ( notice
+  , copyFileVerbose
+  , createDirectoryIfMissingVerbose
+  , rawSystemExit
+  , info
+  )
 import System.Directory (getCurrentDirectory)
 import Distribution.Text (display)
 import Distribution.Simple.Compiler
-import Distribution.System (buildPlatform)
+import Distribution.System (buildPlatform, buildOS, OS(..))
 import Distribution.Simple.BuildPaths
   ( mkGenericSharedBundledLibName
+  , mkGenericSharedLibName
   , mkGenericStaticLibName
   , dllExtension
   )
 
 main :: IO ()
 main =
   defaultMainWithHooks
-    simpleUserHooks
-      { buildHook = rustBuildHook
-      , cleanHook = rustCleanHook
-      }
-
-rustCleanHook
-  :: PD.PackageDescription
-  -> ()
-  -> UserHooks
-  -> CleanFlags
-  -> IO ()
-rustCleanHook desc () hooks flags = do
-  pwd <- getCurrentDirectory
-  putStrLn $ "cleanup rust build artifacts: delete " <> pwd <> "/target"
-  rawSystemExit (fromFlag $ cleanVerbosity flags) "rm" ["-r", "./target"]
-  cleanHook simpleUserHooks desc () hooks flags
```

Comment:
A dedicated clean hook is not needed any more, because the Rust build now happens within the `dist-newstyle` directory, which is deleted by the default clean hook.